packages/language-core/src/generators/template.ts

```diff
@@ -1223,6 +1223,8 @@ export function generate(
 		}
 		codes.push(`}, `);
 
+		const canCamelize = !nativeTags.has(node.tag) || node.tagType === CompilerDOM.ElementTypes.COMPONENT;
```

Comment:
I will see if `|| node.tagType === CompilerDOM.ElementTypes.SLOT` is also need or not. (The previous comment in the wrong line has been deleted.)

Reply:
Just a remainder that my original fix used `node.tagType !== CompilerDOM.ElementTypes.ELEMENT` which would maybe be even more accurate but I guess it has to be tested.

Reply:
Okay we don't need that, because the template AST given slot prop name is already camelize. I believe this has always been an edge issue with mapping on the other side and will investigate when I have time.